### PR TITLE
Remove or change Account and Contact filters and sorting

### PR DESCRIPTION
**Pull request by jjbennett**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/UTIL_Typeahead_TEST.cls [hunk 1]
```diff
@@ -49,11 +49,11 @@ public with sharing class UTIL_Typeahead_TEST {
 	      Test.setFixedSearchResults(fixedSearchResults);
 	
 	    Test.startTest();
-	    list<sObject> accts = UTIL_Typeahead.searchRecords( 'test', 'Account', null, null, null, null, null );
+	    list<sObject> accts = UTIL_Typeahead.searchRecords( 'test', 'Account', null, null, null, 'Id', null );
```
Comment (force2b) [inline src/classes/UTIL_Typeahead_TEST.cls]: Is there a change needed in the `UTIL_Typeahead.component` file as well? I noticed that this method is called from that component passing in a value for the sort parameter, which could be null (or `Name` - the default value in the component).

File: src/classes/CAO_Constants.cls [hunk 2]
```diff
@@ -142,32 +142,32 @@ public with sharing class CAO_Constants {
...
-            if(acct.size()>0){
-                if(acct[0].npe01__SYSTEMISINDIVIDUAL__c==false || acct[0].npe01__SYSTEM_AccountType__c<>CAO_Constants.BUCKET_ORGANIZATION_TYPE){
-                    acct[0].npe01__SYSTEMISINDIVIDUAL__c = true;
-                    acct[0].npe01__SYSTEM_AccountType__c = CAO_Constants.BUCKET_ORGANIZATION_TYPE;
-                    update acct[0];
-                }
-                IndividualAccountId = acct[0].id;
+        if (individualAccountId == null) {
+            List<Account> acct = [
+                SELECT Id, npe01__SYSTEMISINDIVIDUAL__c, npe01__SYSTEM_AccountType__c
```
[the remark this answers is not in the document]
Comment (njjc) [reply src/classes/CAO_Constants.cls]: I don't believe there are any cases where orgs have Individual accounts accessible by Name only, I'm totally fine with this solution. (The code that does this check is over 7 years old)[https://github.com/SalesforceFoundation/Contacts_and_Organizations/blame/master/src/classes/Constants.cls#L165], and its reason for existing is lost in the ether... my guess is that we introduced those SYSTEM fields and used that method to populate them.

File: src/classes/CON_ContactMerge.cls [hunk 3]
```diff
@@ -126,15 +126,22 @@ public class CON_ContactMerge {
         
         //get loser contact Accounts that should be merged out.  This includes
         // One2One accounts and Solo Household Accounts
-        loserAccountsToBeMerged = [Select Id,npe01__SYSTEMIsIndividual__c,npe01__SYSTEM_AccountType__c,Name from Account a
-            where Id IN :loserAccountIds and 
-            (npe01__SYSTEM_AccountType__c=:CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE or
-                (npe01__SYSTEM_AccountType__c=:CAO_Constants.HH_ACCOUNT_TYPE and Number_of_Household_Members__c = 1))
-            and Name<>:CAO_Constants.BUCKET_ACCOUNT_NAME];
```
Comment (RadSFDeveloper) [inline src/classes/CON_ContactMerge.cls]: The check for `and Name<>:CAO_Constants.BUCKET_ACCOUNT_NAME` has been removed here. Can there be an Account with `ONE_TO_ONE_ORGANIZATION_TYPE` or `HH_ACCOUNT_TYPE` but with `CAO_Constants.BUCKET_ACCOUNT_NAME` Name?
@njjc , @force2b

Comment (jjbennett) [reply src/classes/CON_ContactMerge.cls]: The CAO constants attach to bucket account will only attach to an account that meets the system account type criteria so even if we have an Account with the same name it's not going to be used as a bucket account in our code.

**Pull request merged**